src/algorithms/digi/PhotoMultiplierHitDigi.cc

```diff
@@ -63,23 +63,25 @@ void eicrecon::PhotoMultiplierHitDigi::AlgorithmChangeRun() {
 //------------------------
 // AlgorithmProcess
 //------------------------
-std::vector<edm4eic::RawTrackerHit*>
-eicrecon::PhotoMultiplierHitDigi::AlgorithmProcess(std::vector<const edm4hep::SimTrackerHit*>& sim_hits) {
-
+std::vector<edm4eic::MCRecoTrackerHitAssociation*> eicrecon::PhotoMultiplierHitDigi::AlgorithmProcess(
+    std::vector<const edm4hep::SimTrackerHit*>& sim_hits
+    )
+{
         m_log->trace("{:=^70}"," call PhotoMultiplierHitDigi::AlgorithmProcess ");
         struct HitData {
           uint32_t npe;
           double signal;
-          double time;
+          decltype(edm4hep::SimTrackerHitData::time) time;
           dd4hep::Position pos;
+          std::vector<const edm4hep::SimTrackerHit*> mc_hits;
         };
         std::unordered_map<decltype(edm4eic::RawTrackerHitData::cellID), std::vector<HitData>> hit_groups;
         // collect the photon hit in the same cell
         // calculate signal
-        for(const auto& ahit : sim_hits) {
+        for(auto &ahit : sim_hits) {
```

Comment:
Why do we lose `const` here? Well, probably we don't lose it and it's inside the `auto`, but manifest `const auto` is preferred .Also, `AlgorithmProcess` probably needs to take a `const std::vector<>& sim_hits`.

Reply:
No idea, reverted, and now using `const std::vector<>& sim_hits` as suggested.